### ISSUE TITLE
Support for compressed CSV files

### ISSUE THREAD

**Issue by Enchufa2** [hannesmuehleisen/MonetDBLite-R#23]
Most CSV readers (notably, base R readers) support compressed (gz, bzip2...) CSV files transparently. It would be a nice addition to `MonetDBLite::monetdb.read.csv`, because big CSVs are commonly gzip'ed.

**Comment by hannesmuehleisen** [comment]
Yes we had removed this feature because it adds a dependency, which is a pain especially on Windows.

**Comment by Enchufa2** [comment]
And what about allowing an external command to be specified? May this be possible? Example: `data.table::fread`. You can read a compressed CSV as follows:

```r
data.table::fread("zcat somefile.csv.gz")
```

`zcat` is invoked and its output feeds the reader.

**Comment by hannesmuehleisen** [comment]
That sounds pretty good. I am unlikely to implement this at the moment. Happy to review a PR though.

**Comment by Enchufa2** [comment]
And another quick and portable option would be to rely on the `R.utils` package, which implements `gunzip` and the like based on base R (efficiently copying from a `gzfile`, `bzfile`... connection to a file connection). But this would mean adding `R.utils` as a dependency (or porting to MonetDBLite just the relevant code). What do you think?
